File: 557. Reverse Words in a String III/solution.cpp
class Solution {
public:
  string reverseWords(string s) {
    string result = "";
    string word = "";
    for (int index = 0; index < s.size(); index++){
      char letter = s[index];
      if (letter != ' '){
        word = letter + word;
      }
      if (letter == ' ' || index == s.size() - 1){
        result += word;
        if (index != s.size() - 1){
          result += " ";
        }
        word = "";
      }
    }

    return result;
  }
};

File: 557. Reverse Words in a String III/solution.py
class Solution(object):
  def reverseWords(self, s):
    """
    :type s: str
    :rtype: str
    """
    result = ""
    word = ""
    for index, letter in enumerate(s):
      if letter != " ":
        word = letter + word
      if letter == " " or index == len(s) - 1:
        result += word
        if index != len(s) - 1:
          result += " "
        word = ""
    
    return result